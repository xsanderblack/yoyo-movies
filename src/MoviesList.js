import React from 'react';
import Movie from './Movie';

const MoviesList = (props) => {
  const { movies, onRemoveFromFavorits } = props;
  return (
    <div>
      {movies &&
        movies.map(movie => 
          <Movie
            key={movie.id}
            {...movie}
            onRemoveFromFavorits={onRemoveFromFavorits}
          />
        )
      }
    </div>
  );
};

export default MoviesList;
