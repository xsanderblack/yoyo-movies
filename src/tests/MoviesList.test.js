import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import MoviesList from '../MoviesList';

it('MoviesList renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MoviesList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render MoviesList correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<MoviesList />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
