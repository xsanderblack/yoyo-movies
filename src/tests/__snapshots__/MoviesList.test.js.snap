// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MoviesList correctly 1`] = `<div />`;
